### PR TITLE
mu: fix command

### PR DESCRIPTION
**Pull request by MaxSchlueter**
----

### Description

This just adds the command line flag to the `mu` command to specify the personal email addresses.
I think the email addresses specified as aliases should also be included, should I open a new pull req for that?

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by berbiche** [comment]
If I understand correctly, this is only used when a user's `mu` configuration is initialized for the first time?

**Comment by MaxSchlueter** [comment]
> If I understand correctly, this is only used when a user's `mu` configuration is initialized for the first time?

Yes, this command is only run once to setup the database.

**Comment by teto** [comment]
I dont use `mu`, how does this PR change the behavior ? `mu init` init for all addresses by default while you want to do it just for accounts with mu.enable ?

**Comment by MaxSchlueter** [comment]
Yes, but you can also specify your personal addresses via `--my-address` option on the command line. See man page for one advantage of doing that: https://manpages.debian.org/experimental/maildir-utils/mu-init.1.en.html

**Comment by MaxSchlueter** [comment]
@KarlJoad since you are the module maintainer, what do you think of this PR?

**Comment by KarlJoad** [comment]
@MaxSchlueter, the only problem with the way it is currently set-up within [`mu.nix`](https://github.com/nix-community/home-manager/blob/master/modules/programs/mu.nix) is that *every* email account with `mu.enable = true` will be a "personal" email account. The expressions that generate that string should really be dropped from the module entirely, but that's a different discussion.

```nix
# Takes the list of accounts with mu.enable = true, and generates a
# command-line flag for initializing the mu database.
myAddresses = let
  # List of account sets where mu.enable = true.
  muAccounts =
    filter (a: a.mu.enable) (attrValues config.accounts.email.accounts);
  addrs = map (a: a.address) muAccounts;
  # Prefix --my-address= to each account's address with mu.enable.
  addMyAddress = map (addr: "--my-address=" + addr) addrs;
in concatStringsSep " " addMyAddress;
```

As you can see, `addrs` is just the list of email accounts where `mu.enable = true`. 

What you really want is a way to optionally select some email accounts to use `mu`, and of those that **do** have `mu` enabled, should allow you to specify an `mu.mainAddress` (or a similar-ish name). Only if both `mu.enable = true` **and** `mu.mainAddress = true` then should you have that email address be added to the `mu init --my-address=<email-address>` command run during generational-switch.

If you want to add the `--my-address` functionality as the way DJCB meant it to work, you will need to perform some additional work.

**Comment by MaxSchlueter** [comment]
@KarlJoad Ok, thank you for the explanation. I thought that every email account specified in a hm config would be a personal email account (or what do you mean by "personal"?). So you want to filter the email accounts for an additional option `mu.mainAddress`, how would you describe that option?

I'm actually not sure how DJCB meant it to work. My use case here actually hasn't anything to do with what is mentioned in the man page, I find it useful because mu4e will let me choose from my personal email addresses when I compose a new message (`mu4e-compose-new`).

**Comment by KarlJoad** [comment]
So, I have figured out what both I and DJCB meant by "personal".  The `man` page for `mu-init` was a little vague on what it does, but I have figured out that `mu-cfind` is meant to search for ***contacts*** within your contacts database and the emails that you have sent/received. The use of the `--personal` flag in that command is meant to filter for only emails that use your email addresses (which are all the ones you specify with the `${myAddresses}` variable. Disregard what I said in https://github.com/nix-community/home-manager/pull/1623#issuecomment-759931036.
```console
--my-address=<my-email-address>

    specifies that some e-mail addresses are 'my-address' (--my-address can be used multiple times).
    This is used by mu cfind -- any e-mail address found in the address fields of a message which also
    has <my-email-address> in one of its address fields is considered a personal e-mail address. This
    allows you, for example, to filter out (mu cfind --personal) addresses which were merely seen in
    mailing list messages.
```

To initialize the database with `mu init`, the `${myAddresses}` is not required to be passed to successfully initialize the database, but it is heavily recommended to do so.

To see the difference, in a safe location, run `mu init --maildir=<path>`, then `mu index`. You'll notice that "personal addresses" returns `<none>`, although the database will still work. However, `mu cfind --personal` will fail (as the `personal` contacts don't exist). Then run `mu init --maildir=<path> --my-address=<address>`, then `mu index`. Then you'll be able to search for contacts using `mu cfind --personal`.

In short, your PR **is necessary** for full `mu` functionality to be present and functioning properly with home-manager. @teto please merge this ASAP.

**Comment by teto** [comment]
No need for the ASAP.

**Comment by KarlJoad** [comment]
Agreed, I'm sorry. Got caught up in a couple of other things. Wrote something there I meant for something else. Apologies.
